### PR TITLE
Monitoring: Switch to using mounted configmaps and use library images

### PR DESCRIPTION
**Pull request by weikinhuang**
## Description
This changes the standard `kubectl` `.yml` deployments to use mounted `ConfigMap`s for setting up alertmanager and prometheus. Deprecating the need for custom images of those services. The helm charts already use `ConfigMap`s.

## Motivation and Context

This gets us one step closer to having less images to maintain. Provides feature parity with https://github.com/openfaas/faas/issues/282 and eases transition for https://github.com/openfaas/faas/issues/283.

## How Has This Been Tested?

Tested by deploying `faas` using the readme in additional to the new configmap.

```
NAME                              READY     STATUS    RESTARTS   AGE
po/alertmanager-cb8f57c6d-4xqfk   1/1       Running   0          3s
po/faas-netesd-5b9db8777d-k9qlf   1/1       Running   0          3s
po/gateway-799dfdd695-mpfjg       1/1       Running   0          3s
po/prometheus-75b55cf4db-ml9xv    1/1       Running   0          3s

NAME               CLUSTER-IP     EXTERNAL-IP   PORT(S)          AGE
svc/alertmanager   10.3.245.203   <nodes>       9093:30010/TCP   3s
svc/faas-netesd    10.3.37.154    <nodes>       8080:31111/TCP   3s
svc/gateway        10.3.146.75    <nodes>       8080:30002/TCP   3s
svc/kubernetes     10.3.0.1       <none>        443/TCP          7m
svc/prometheus     10.3.61.248    <nodes>       9090:30009/TCP   3s

NAME                  DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deploy/alertmanager   1         1         1            1           3s
deploy/faas-netesd    1         1         1            1           3s
deploy/gateway        1         1         1            1           3s
deploy/prometheus     1         1         1            1           3s

NAME                        DESIRED   CURRENT   READY     AGE
rs/alertmanager-cb8f57c6d   1         1         1         3s
rs/faas-netesd-5b9db8777d   1         1         1         3s
rs/gateway-799dfdd695       1         1         1         3s
rs/prometheus-75b55cf4db    1         1         1         3s
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by weikinhuang** [comment]
Configs pulled from https://github.com/openfaas/faas/tree/master/prometheus

**Comment by alexellis** [comment]
Thank you. Please can you confirm auto-scaling still works and all metrics are present before I merge?

File: README.md [hunk 1]
```diff
@@ -82,10 +82,10 @@ $ minikube start --vm-driver=xhyve
 **Deploy FaaS-netes and the API Gateway**
 
 ```
-$ kubectl apply -f ./faas.yml,monitoring.yml
+$ kubectl apply -f ./faas.yml,monitoring-config.yml,monitoring.yml
```
Comment (alexellis) [inline README.md]: should we merge this into one?

Comment (weikinhuang) [reply README.md]: Prevents having to duplicate the config for arm and non arm

**Comment by weikinhuang** [comment]
Tested autoscaling
```bash
ip=$(minikube ip); while [ true ] ; do curl $ip:31112/function/nodeinfo -d "" ; done
```

```
$ kubectl get all
NAME                              READY     STATUS    RESTARTS   AGE
po/alertmanager-cb8f57c6d-4nczg   1/1       Running   0          5m
po/faas-netesd-5b9db8777d-t2cmp   1/1       Running   0          5m
po/gateway-799dfdd695-bng99       1/1       Running   0          5m
po/nodeinfo-d466dd64c-9qfnw       1/1       Running   0          8s
po/nodeinfo-d466dd64c-dkt6t       1/1       Running   0          8s
po/nodeinfo-d466dd64c-pw75m       1/1       Running   0          8s
po/nodeinfo-d466dd64c-vpfs2       1/1       Running   0          27m
po/nodeinfo-d466dd64c-x9jv2       1/1       Running   0          8s
po/prometheus-75b55cf4db-24dvz    1/1       Running   0          5m

NAME               CLUSTER-IP     EXTERNAL-IP   PORT(S)          AGE
svc/alertmanager   10.3.131.97    <nodes>       9093:31113/TCP   5m
svc/faas-netesd    10.3.232.173   <nodes>       8080:31111/TCP   5m
svc/gateway        10.3.237.18    <nodes>       8080:30002/TCP   5m
svc/kubernetes     10.3.0.1       <none>        443/TCP          35m
svc/nodeinfo       10.3.12.31     <none>        8080/TCP         27m
svc/prometheus     10.3.28.215    <nodes>       9090:31119/TCP   5m

NAME                  DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deploy/alertmanager   1         1         1            1           5m
deploy/faas-netesd    1         1         1            1           5m
deploy/gateway        1         1         1            1           5m
deploy/nodeinfo       5         5         5            5           27m
deploy/prometheus     1         1         1            1           5m

NAME                        DESIRED   CURRENT   READY     AGE
rs/alertmanager-cb8f57c6d   1         1         1         5m
rs/faas-netesd-5b9db8777d   1         1         1         5m
rs/gateway-799dfdd695       1         1         1         5m
rs/nodeinfo-d466dd64c       5         5         5         27m
rs/prometheus-75b55cf4db    1         1         1         5m
```

**Comment by weikinhuang** [comment]
And scaled back down after a minute

```bash
$ kubectl get all
NAME                              READY     STATUS    RESTARTS   AGE
po/alertmanager-cb8f57c6d-4nczg   1/1       Running   0          6m
po/faas-netesd-5b9db8777d-t2cmp   1/1       Running   0          6m
po/gateway-799dfdd695-bng99       1/1       Running   0          6m
po/nodeinfo-d466dd64c-vpfs2       1/1       Running   0          28m
po/prometheus-75b55cf4db-24dvz    1/1       Running   0          6m

NAME               CLUSTER-IP     EXTERNAL-IP   PORT(S)          AGE
svc/alertmanager   10.3.131.97    <nodes>       9093:31113/TCP   6m
svc/faas-netesd    10.3.232.173   <nodes>       8080:31111/TCP   6m
svc/gateway        10.3.237.18    <nodes>       8080:30002/TCP   6m
svc/kubernetes     10.3.0.1       <none>        443/TCP          36m
svc/nodeinfo       10.3.12.31     <none>        8080/TCP         28m
svc/prometheus     10.3.28.215    <nodes>       9090:31119/TCP   6m

NAME                  DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deploy/alertmanager   1         1         1            1           6m
deploy/faas-netesd    1         1         1            1           6m
deploy/gateway        1         1         1            1           6m
deploy/nodeinfo       1         1         1            1           28m
deploy/prometheus     1         1         1            1           6m

NAME                        DESIRED   CURRENT   READY     AGE
rs/alertmanager-cb8f57c6d   1         1         1         6m
rs/faas-netesd-5b9db8777d   1         1         1         6m
rs/gateway-799dfdd695       1         1         1         6m
rs/nodeinfo-d466dd64c       1         1         1         28m
rs/prometheus-75b55cf4db    1         1         1         6m
```

File: README.md (hunk 1 above)
Comment (alexellis) [reply README.md]: OK makes sense. This will also need updating on the OpenFaaS K8s deployment guide.

Comment (weikinhuang) [reply README.md]: I see it, will do

**Comment by alexellis** [comment]
Very excited to merge this. @elliott-beach @zouyee do you have time to kick the tyres with this?

File: README.md (hunk 1 above)
Comment (alexellis) [reply README.md]: Hi Wei did you update the guide?

Comment (alexellis) [reply README.md]: (this is in the main FaaS repo) https://github.com/openfaas/faas/blob/master/guide/deployment_k8s.md

Comment (weikinhuang) [reply README.md]: yes it's in https://github.com/openfaas/faas/pull/287

File: monitoring.armhf.yml [hunk 2]
```diff
@@ -27,12 +27,30 @@ spec:
     spec:
       containers:
       - name: prometheus
-        image: alexellis2/prometheus-armhf:1.5.2-k8s # alexellis2/prometheus-armhf:1.5.2
+        image: alexellis2/prometheus-armhf:1.5.2
```
Comment (zouyee) [inline monitoring.armhf.yml]: why not using `prom/prometheus`?

**Comment by zouyee** [comment]
It may be a better choose [prometheus-operator](https://github.com/coreos/prometheus-operator)

**Comment by johnmccabe** [comment]
@alexellis I'll test this in the morning, I expect it to work as I ended up making the same changes in #79 before I'd realised it had already been done here.

File: README.md (hunk 1 above)
Comment (johnmccabe) [inline README.md]: Can you add the `rbac.yml` here and get rid of the optional section. RBAC isn't optional now surely?

Comment (weikinhuang) [reply README.md]: I think we should make that change in a upcoming PR instead. I feel that PRs should only contain singular feature changes. That way things are easier to track. Especially if things break, it becomes much easier to revert. I believe there's also a discussion in the issues section about the changes in k8s 1.8 with the rbac api version.

Comment (johnmccabe) [reply README.md]: Ok, fair enough.

**Comment by johnmccabe** [comment]
Apart from a rebase, this is looking good to me. Have spun up both async and non-async deployments and confirmed autoscaling up/down works as before.

👍

**Comment by johnmccabe** [comment]
@alexellis can we get this merged

**Comment by weikinhuang** [comment]
rebased

**Comment by alexellis** [comment]
We have mounted configs now @weikinhuang thank you for doing this work.

**Pull request closed without merging**